--- origsrc/liblouisxml-2.4.0/liblouisxml/louisxml.h	2011-05-11 07:12:14.000000000 -0500
+++ src/liblouisxml-2.4.0/liblouisxml/louisxml.h	2013-10-30 20:24:30.068924000 -0500
@@ -300,5 +300,12 @@ int start_style (StyleType * curStyle);
 int end_style (StyleType * curStyle);
 int find_action (const char **actions, const char *action);
 int find_group_length (const char groupSym[2], const char *groupStart);
+void unsignedcharcpy(char* to, const char* from, int length);
+void widecharcpy(widechar* to, const widechar* from, int length);
+void widestrcpy(widechar* to, const widechar* from);
+int do_pagenum (void);
+int writeBuffer(int from, int skip);
+int savePointers(void);
+int restorePointers(void);
 
 #endif /*louisxml_h */
--- origsrc/liblouisxml-2.4.0/liblouisxml/makeContents.c	2011-05-11 07:12:14.000000000 -0500
+++ src/liblouisxml-2.4.0/liblouisxml/makeContents.c	2013-10-30 20:19:14.304863300 -0500
@@ -63,9 +63,9 @@ initialize_contents (void)
 {
 
   saved_braillePageNumberFormat = ud->brl_page_num_format;
-  widestrcpy(saved_printPageNumber, ud->print_page_number, -1);
-  widestrcpy(saved_printPageNumberFirst, ud->print_page_number_first, -1);
-  widestrcpy(saved_printPageNumberLast, ud->print_page_number_last, -1);
+  widestrcpy(saved_printPageNumber, ud->print_page_number);
+  widestrcpy(saved_printPageNumberFirst, ud->print_page_number_first);
+  widestrcpy(saved_printPageNumberLast, ud->print_page_number_last);
   ud->after_contents = 1;
   saved_udContents = ud->contents;
   saved_linesOnPage = ud->lines_on_page;
@@ -191,9 +191,9 @@ make_contents (void)
       styleSpec = &ud->style_stack[ud->style_top];
       styleSpec->curBrlNumFormat = saved_braillePageNumberFormat;
       ud->brl_page_num_format = saved_braillePageNumberFormat;
-      widestrcpy(ud->print_page_number, saved_printPageNumber, -1);
-      widestrcpy(ud->print_page_number_first, saved_printPageNumberFirst, -1);
-      widestrcpy(ud->print_page_number_last, saved_printPageNumberLast, -1);
+      widestrcpy(ud->print_page_number, saved_printPageNumber);
+      widestrcpy(ud->print_page_number_first, saved_printPageNumberFirst);
+      widestrcpy(ud->print_page_number_last, saved_printPageNumberLast);
       do_newpage ();
       ud->contents = 2;
       currentHeading = firstHeading;
--- origsrc/liblouisxml-2.4.0/liblouisxml/semantics.c	2011-05-11 07:12:14.000000000 -0500
+++ src/liblouisxml-2.4.0/liblouisxml/semantics.c	2013-10-30 20:20:28.041080800 -0500
@@ -33,6 +33,7 @@
 #include <string.h>
 #include "louisxml.h"
 #include <libxml/xpath.h>
+#include <libxml/xpathInternals.h>
 #include "sem_names.h"
 
 typedef struct
